cpp/src/io/comp/comp.cpp

```diff
@@ -339,29 +339,34 @@ void host_compress(compression_type compression,
     return h_inputs[a].size() > h_inputs[b].size();
   });
 
-  std::vector<std::future<size_t>> tasks;
+  auto h_results =
+    cudf::detail::make_pinned_vector_async<compression_result>(results.size(), stream);
+  cudf::detail::cuda_memcpy<compression_result>(h_results, results, stream);
+
+  std::vector<std::future<std::pair<size_t, size_t>>> tasks;
   auto const num_streams =
     std::min<std::size_t>(num_chunks, cudf::detail::host_worker_pool().get_thread_count());
   auto const streams = cudf::detail::fork_streams(stream, num_streams);
```

Comment:
Pardon my ignorance, but shouldn't there be an associated call to `cudf::detail::join_streams` at the end?  I can't seem to find it.

Reply:
My understanding is the end of each task includes a call to `cudf::detail::cuda_memcpy` which does a `stream.synchronize()` so each task/stream should be finished when `.get()` is called. So there should be no need to call `join_streams` but I could be wrong.

Reply:
That's exactly right. There's no async work in the streams after the futures return.